File: euler78.c
#include <stdio.h>

#define BOUND 100000

int main() {
	int ways[BOUND+1];
	int i, j;
	ways[0] = 1;
	for (i = 1; i <= BOUND; i++) {
		for (j = i; j <= BOUND; j++) 
			ways[j] = (ways[j] + ways[j-i]) % 1000000;
		if (ways[i] == 0) {
			printf("%d has %d (partitions %% 1000000)\n", i, ways[i]);
			break;
		}
		if (i % 1000 == 0)
			printf("%d\n",i);
	}
	return 0;
}

File: euler78.py
bound = 50000
ways = [0 for x in range(bound+1)]
ways[0] = 1

for i in range(1, bound+1):#for each number that a sum could be split into
	for j in range(i, bound+1):#for each number that can include the largest number we are up to in the sum
		ways[j] = (ways[j] + ways[j-i]) % 1000000#the first part of the sum = i, the number of ways to make j += the number of ways to make j-i
	if ways[i] == 0:
		print "%d has %d (partitions % 1000000)" % (i, ways[i])
		break
	if i % 1000 == 0:
		print i

print ways[bound]